src/sqlancer/common/query/SQLQueryAdapter.java

```diff
@@ -96,9 +96,19 @@ public String getUnterminatedQueryString() {
     }
 
     public void checkException(Exception e) throws AssertionError {
-        if (!expectedErrors.errorIsExpected(e.getMessage())) {
-            throw new AssertionError(query, e);
+        Throwable ex = e;
+
+        while (true) {
```

Comment:
Thanks for the contribution. I think the code should be easier to understand if you use `ex.getCause() != null` rather than `true` in the `while` condition, which would also simplify the `else if` to an `else`. Or would this yield an unexpected result?

Reply:
Agreed, it's overcomplicated a bit :) Thanks

This is simplified version
```
Throwable ex = e;

while (ex != null) {
    if (expectedErrors.errorIsExpected(ex.getMessage())) {
        return;
    } else {
        ex = ex.getCause();
    }
}

throw new AssertionError(query, e);
```